app/src/main/java/fr/free/nrw/commons/di/FixedDaggerAppCompatActivity.java

```diff
@@ -0,0 +1,43 @@
+package fr.free.nrw.commons.di;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AppCompatActivity;
+
+import javax.inject.Inject;
+
+import dagger.android.AndroidInjector;
+import dagger.android.DispatchingAndroidInjector;
+import dagger.android.support.HasSupportFragmentInjector;
+
+public abstract class FixedDaggerAppCompatActivity extends AppCompatActivity implements HasSupportFragmentInjector {
```

Comment:
Adding "Fixed" to the name is a reaction to our current problems but our "future selves" (6 months / a year down the line) wont remember the context as to why its "fixed" or what it actually fixed.  Personally speaking, I would drop the "Fixed" part of the name in favor of (say) `CommonsDaggerAppCompatActivity`.  In addition to a rename, it would be helpful to add a comment as to why we need this.

Reply:
Nice point! Renaming.